be/src/storage/memtable_flush_executor.cpp

```diff
@@ -23,30 +23,47 @@
 
 #include <memory>
 
+#include "gen_cpp/data.pb.h"
 #include "runtime/current_thread.h"
 #include "storage/memtable.h"
 
 namespace starrocks {
 
 class MemtableFlushTask final : public Runnable {
 public:
-    MemtableFlushTask(FlushToken* flush_token, std::unique_ptr<vectorized::MemTable> memtable)
-            : _flush_token(flush_token), _memtable(std::move(memtable)) {}
+    MemtableFlushTask(FlushToken* flush_token, std::unique_ptr<vectorized::MemTable> memtable, bool eos,
+                      std::function<void(std::unique_ptr<SegmentPB>, bool)> cb)
+            : _flush_token(flush_token), _memtable(std::move(memtable)), _eos(eos), _cb(std::move(cb)) {}
 
     ~MemtableFlushTask() override = default;
 
     void run() override {
-        SCOPED_THREAD_LOCAL_MEM_SETTER(_memtable->mem_tracker(), false);
-
-        _flush_token->_stats.cur_flush_count++;
-        _flush_token->_flush_memtable(_memtable.get());
-        _flush_token->_stats.cur_flush_count--;
-        _memtable.reset();
+        std::unique_ptr<SegmentPB> segment = nullptr;
+        if (_memtable) {
+            SCOPED_THREAD_LOCAL_MEM_SETTER(_memtable->mem_tracker(), false);
+            segment = std::make_unique<SegmentPB>();
+
+            _flush_token->_stats.cur_flush_count++;
+            _flush_token->_flush_memtable(_memtable.get(), segment.get());
```

Comment:
memtable may be nullptr here, call memtable->flush will crash be

Reply:
already judge in line:42?